servicetalk-http-api/src/main/java/io/servicetalk/http/api/CharSequences.java

```diff
@@ -170,30 +118,41 @@ public static boolean contentEquals(final CharSequence a, final CharSequence b)
      * @return The index of {@code c} or {@code -1} otherwise.
      */
     public static int indexOf(CharSequence sequence, char c, int fromIndex) {
-        if (sequence instanceof String) {
-            return ((String) sequence).indexOf(c, fromIndex);
-        } else if (sequence instanceof AsciiBuffer) {
-            return ((AsciiBuffer) sequence).indexOf(c, fromIndex);
-        }
-        for (int i = fromIndex; i < sequence.length(); ++i) {
-            if (sequence.charAt(i) == c) {
-                return i;
-            }
-        }
-        return -1;
+       return io.servicetalk.buffer.api.CharSequences.indexOf(sequence, c, fromIndex);
+    }
+
+    /**
+     * Calculate a hash code of a byte array assuming ASCII character encoding.
+     * The resulting hash code will be case insensitive.
+     *
+     * @param seq The ascii string to produce hashcode for.
+     * @return a hashcode for the given input.
+     */
+    public static int caseInsensitiveHashCode(CharSequence seq) {
+        return io.servicetalk.buffer.api.CharSequences.caseInsensitiveHashCode(seq);
+    }
+
+    /**
+     * Split a given {@link CharSequence} to separate ones on the given {@code delimiter}.
+     * The returned {@link CharSequence}s are created by invoking the {@link CharSequence#subSequence(int, int)} method
+     * on the main one.
+     *
+     * This method has no support for regex.
+     *
+     * @param input The initial {@link CharSequence} to split, this experiences no side effects
+     * @param delimiter The delimiter character
+     * @return a {@link List} of {@link CharSequence} subsequences of the input with the separated values
+     */
+    public static List<CharSequence> split(final CharSequence input, final char delimiter) {
```

Comment:
Not necessary to add new methods for this deprecated class, consider delegating only methods that exist in this class

Reply:
Split was already there before.